domain/service/servicestore.go

```diff
@@ -242,40 +299,52 @@ func (s *storeImpl) FindTenantByDeploymentID(ctx datastore.Context, deploymentID
 
 	if results.Len() == 0 {
 		return nil, nil
-	} else if svcs, err := convert(results); err != nil {
+	} else if svcs, err := s.convert(results); err != nil {
 		return nil, err
 	} else {
 		return &svcs[0], nil
 	}
 }
 
-func query(ctx datastore.Context, query string) ([]Service, error) {
+func (s *storeImpl) query(ctx datastore.Context, query string) ([]Service, error) {
 	q := datastore.NewQuery(ctx)
 	elasticQuery := search.Query().Search(query)
 	search := search.Search("controlplane").Type(kind).Size("50000").Query(elasticQuery)
 	results, err := q.Execute(search)
 	if err != nil {
 		return nil, err
 	}
-	return convert(results)
+	return s.convert(results)
 }
 
+// fillAdditionalInfo fills the service object with additional information
+// that amends or overrides what was retrieved from elastic
+func (s *storeImpl) fillAdditionalInfo(svc *Service) {
+	plog.WithFields(log.Fields{
+		"serviceId": svc.ID,
+		"serviceName": svc.Name,
+	}).Debug("Adding additional info to Elastic result")
+	fillConfig(svc)
+	s.fillVolatileInfo(svc) // Uses RLock
+}
+
+// fillConfig fills in the ConfgiFiles values
 func fillConfig(svc *Service) {
```

Comment:
For consistency with a lot of the other changes in this file, I would make `fillConfig()` a method on `storeImpl` as well